pkg/api/vmtemplate/store.go

```diff
@@ -0,0 +1,138 @@
+package vmtemplate
+
+import (
+	"sync"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/rancher/apiserver/pkg/apierror"
+	"github.com/rancher/apiserver/pkg/types"
+	ctlvmv1alpha1 "github.com/rancher/harvester/pkg/generated/controllers/vm.cattle.io/v1alpha1"
+	"github.com/rancher/harvester/pkg/ref"
+	"github.com/rancher/wrangler/pkg/schemas/validation"
+)
+
+const (
+	templateLabel = "template.vm.cattle.io/templateID"
+)
+
+type templateVersionStore struct {
+	types.Store
+	templateCache        ctlvmv1alpha1.TemplateCache
+	templateVersions     ctlvmv1alpha1.TemplateVersionClient
+	templateVersionCache ctlvmv1alpha1.TemplateVersionCache
+	mu                   sync.Mutex
```

Comment:
can u add a comment to explain why and how the `sync.Mutex` is used.

Reply:
move to controller and add comment their.